### DISCUSSION TITLE
First row of self referencing table

### DISCUSSION THREAD

**Discussion by TheCataliasTNT2k** [SeaQL/sea-orm#1354] [Q&A] [answered]
How can I add the first row to a table, that has a self referencing column?
There is an example, how to create such a column: https://www.sea-ql.org/SeaORM/docs/relation/self-referencing/
But how to insert the first row into that table?
I can not reference a row if there is none, and if I try to reference the row by itself, I get an error:
`Execution Error: error returned from database: (code: 1) foreign key mismatch - "user" referencing "user"`

**Comment by billy1624** [comment, marked as answer]
Hey @TheCataliasTNT2k, let say we have a table with `id` column and a foreign key column `id_ref`. The `id_ref` column must be nullable. So that we can insert a row that you want to reference first. Then, update the value of `id_ref` column. There you have it, a self referencing row.

**Comment by TheCataliasTNT2k** [comment]
Found the answer:
I accidentally created my relation the wrong way around.
Thank you nevertheless!
